src/LIS2DW12Sensor.h

```diff
@@ -44,7 +44,7 @@
 
 
 /* Includes ------------------------------------------------------------------*/
-
+#include <Arduino.h>
 #include "Wire.h"
 #include "SPI.h"
```

Comment:
`SPI.h` includes `Arduino.h` which includes `wiring.h` including `wiring_constants.h`. `MSB_FIRST `is defined here.
So I don't know how you got this issue but I'm not agree with this change. I guess this workaround hide something weird in PIO env.

Reply:
Well, while I must agree with you and can't dispute your logic, kind sir.

Let me propose a simple test:

```
mkdir Deleteme && cd Deleteme
pio init -b esp32dev --ide vscode
cat <<EOT >> src/main.cpp
#include <Arduino.h>
#include <LIS2DW12Sensor.h>

LIS2DW12Sensor Test(&Wire);

void setup() {
  Serial.begin(115200);
}

void loop() {
}
EOT

cat <<EOT >> platformio.ini
lib_deps =
 https://github.com/stm32duino/LIS2DW12.git
EOT

pio run 
```

This will fail with the previously described undefined message, on my side.

Reply:
Ah I better understand.
It is with another core than the STM32 one. You used ESP32.
So subject is not to be PIO compliant but ESP32.

We do not tested against esp32. This library currently supported those arch:
https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_index.json

Reply:
Oh, I see. Well, I'm using it on ESP32, and this change is enough to make it also work on ESP32, thanks to the amazing framework compatibilities provided by Arduino.

I would understand if you can't include support for ESP32 since you didn't test against it, but I can confirm I've tested this library with the espressif board, and it works wonders.

Reply:
Looking at the ESP32, as their SPI.h does not include Arduino.h then you have this issue because `MSBFIRST `is defined by it:
https://github.com/espressif/arduino-esp32/blob/f1b06d2a74acdef190ac16e83d48ab619f557e73/cores/esp32/Arduino.h#L51

I think we do not have this issue with Arduino IDE as it automatically adds Arduino.h on top of the sketch.

So, could you update this PR to reference esp32 in the library properties.

@cparata are you agreed with this?

Reply:
@fpistm , done! Sorry for the delay.

Reply:
Thanks.